api/src/utils/common-challenge-functions.ts

```diff
@@ -123,15 +123,13 @@ export async function updateUserChallengeData(
     : null;
 
   if (alreadyCompleted && oldChallenge) {
-    const challengeWithOldDate = userCompletedChallenges[oldIndex];
-
-    if (challengeWithOldDate) {
-      challengeWithOldDate.completedDate = oldChallenge.completedDate;
-      finalChallenge = {
-        ...completedChallenge,
-        ...challengeWithOldDate
-      };
-    }
+    finalChallenge = {
+      ...oldChallenge, // Do we need to do this as the old code just directly spreads the new challenge data
```

Comment:
Do we need to do this as the old code just directly spreads the new challenge data? I don't remember why we did this. 

@Sembauke maybe you might know?

Reply:
I don't think so, no. Since this directly affects what goes in the db, it should just do what the api-server does.

Reply:
Thanks for the confirmation